conda_smithy/ci_register.py

```diff
@@ -178,36 +178,54 @@ def add_project_to_travis(user, project):
     headers = travis_headers()
     endpoint = 'https://api.travis-ci.org'
 
-    url = '{}/repos/{user}/{project}'.format(endpoint, user=user, project=project)
-
-    count = 0
-
-    while True:
-        count += 1
-
+    def get_repo_info():
+        url = '{}/repos/{user}/{project}'.format(endpoint, user=user, project=project)
         response = requests.get(url, headers=headers)
         try:
             response.raise_for_status()
             content = response.json()
             if "repo" in content:
-                break
+                return content
         except requests.HTTPError as e:
-            # We regularly seem to hit this issue during automated feedstock registration on github.
-            # https://github.com/conda-forge/conda-smithy/issues/233
-            # Repo is not available on travis-ci yet.
             print(e)
-
-        if count == 1:
+        return {}
+
+    def wait_until_synced(ignore=False):
+        is_sync_url = '{}/users'.format(endpoint)
+        response = requests.post(is_sync_url, headers=headers)
+        content = response.json()
+        for c in range(20):
+            if ("is_syncing" in content and content["is_syncing"] == False):
+                break
+            time.sleep(3)
+        else:
+            if ignore:
+                print(" * Travis is being synced by somebody else. Ignoring")
+            else:
+                raise RuntimeError('Synching is not finishing for a minute now')
```

Comment:
Any thoughts on breaking these closures out as functions? Personally have nothing against closures. Just thinking this may allow for these to be more reusable and keep the code compact here.